### PR TITLE
Allows override query relation in descendant class

### PR DESCRIPTION
**Pull request by leandrogehlen**
Moved from #13441

### CODE REVIEW

File: framework/db/BaseActiveRecord.php [hunk 1]
```diff
@@ -411,12 +405,25 @@ public function hasOne($class, $link)
...
     public function hasMany($class, $link)
     {
+        return $this->createRelationQuery($class, $link, true);
+    }
+
+    /**
+     * Creates query instance for `has-one` and `has-many` relation.
+     * @param string $class the class name of the related record
+     * @param array $link the primary-foreign key constraint
+     * @param boolean whether this query represents a relation to more than one record.
```
Comment (samdark) [inline framework/db/BaseActiveRecord.php]: Variable name `$multiple` missing.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Need a line for changelog.

**Review by dynasource — APPROVED** [review]

**Review by SilverFire — APPROVED** [review]
:exclamation:  CHANGELOG should be adjusted before merge

**Comment by cebe** [comment]
what was the use case for this? its not clear to me from #13441

**Comment by leandrogehlen** [comment]
@cebe the relation queries are been created by `find` method, then if  i change the default behavior in a `MultiTenantRecord` this changes will happens in relational query.

Example: 

```php
class MultiTenantRecord extends ActiveRecord
{
    public static function find()
    {
        $identity = Yii::$app->user->identity;
       if ($identity instanceof TenantInterface) {
           return parent::find()->where([static::tableName().'.tenant_id' => $identity->getTenantId()]);
       } else {
           throw new NotSupportedException("Identity does not implements TenantInteface");
       }
    }
}

class Category extends MultiTenantRecord 
{
    ...
}
class Product extends MultiTenantRecord 
{
    public function getCategory()
    {
        return $this->hasOne(Category ::className(), ['id' => 'category_id']);
    }
}

Product::find()->joinWith('category')->all();// this generate something like:
```
```sql
select 
  * 
from product 
join category on product.category_id = category.id 
where product.tenant_id = 1 
  and category.tenant_id = 1 -- this is not necessary
```

**Review by cebe — APPROVED** [review]

**Comment by cebe** [comment]
merged, thank you!

**Pull request closed without merging**